File: csharp/csvviewer/csvviewer/csvviewer/CommandLine.cs
namespace csvviewer
{
    public class CommandLine
    {
        private int _pageLength;

        public string GetFilename(string[] args) {
            return args[0];
        }

        public int GetPageLength(string[] args) {
            _pageLength = args.Length > 1 ? int.Parse(args[1]) : 10;
            return _pageLength;
        }

        public int GetPageLength() {
            return _pageLength;
        }
    }
}

File: cpp/csvviewer/csvviewer/Commandline.cpp
#include "Commandline.h"

namespace csvviewer
{
    std::string Commandline::GetFilename(const std::vector<std::string>& args)
    {
        if(args.empty())
            return "persons.csv";
        return args[0];
    }

    int Commandline::GetPageLength(const std::vector<std::string>& args)
    {
        if(args.size() > 1)
            m_pageLength = std::stoi(args[1]);
        return m_pageLength;
    }

    int Commandline::GetPageLength() const
    {
        return m_pageLength;
    }
}